boards/arm/beagle_bbai64_r5/doc/index.rst

```diff
@@ -0,0 +1,96 @@
+.. _beaglebone_ai64:
+
+BeagleBone AI-64
+################
+
+Overview
+********
+BeagleBone AI-64 is a computational platform powered by TI J721E Soc, which is
+targeted for automotive applications.
+
+.. figure:: assets/bbai_64.png
+		:align: center
+		:width: 500px
+		:alt: BeagleBoard.org BeagleBone AI-64
+
+Hardware
+********
+The BeagleBone AI-64 is powered by TI J721E Soc, which has three domains (Main,
+MCU, WKUP). This document gives overview of Zephyr running on Cortex R5 in the
+Main domain.
+
+L1 Memory System
+----------------
+* 16 KB instruction cache.
+* 16 KB data cache.
+* 64 KB TCM.
+
+Region Address Translation
+--------------------------
+The RAT module performs a region based address translation. It translates a
+32-bit input address into a 48-bit output address. Any input transaction that
+starts inside of a programmed region will have its address translated, if the
+region is enabled.
+
+VIM Interrupt Controller
+------------------------
+The VIM aggregates device interrupts and sends them to the R5F CPU(s). The VIM
+module supports 512 interrupt inputs per R5F core. Each interrupt can be either
+a level or a pulse (both active-high). The VIM has two interrupt outputs per core
+IRQ and FIQ.
+
+Supported Features
+******************
+The board configuration supports,
+
++-----------+------------+----------------------+
+| Interface | Controller | Driver/Component     |
++===========+============+======================+
+| UART      | on-chip    | serial               |
++-----------+------------+----------------------+
+
+Other hardwares features are currently not supported.
+
+The default configuration can be found in the defconfig file:
+
+``boards/arm/beagle_bbai64_r5/beagle_bbai64_r5_defconfig``
+
+Flashing
+********
+The Zephyr image can be flashed to J721E Cortex R5 through remoteproc from
+Linux (Running on A72).
+
+To build hello world application for BeagleBone AI-64:
+
+``west build -p auto -b beagle_bbai64_r5 samples/hello_world``
+
+Note:
+-----
+Use the latest Linux image from https://beagleboard.org/latest-images to bootup
+the BBAI-64. This is required for loading zephyr.elf/binary using remoteproc.
+
+Steps to flash the image
+------------------------
+The Example shows how to load an image on Cortex R5FSS0_CORE0 on J721e.
+
+| Copy Zephyr image to the /lib/firmware/ directory.
+| ``cp build/zephyr/zephyr.elf /lib/firmware/``
+|
+| Ensure the Core is not running.
+| ``echo stop > /sys/class/remoteproc/remoteproc18/state``
+|
+| Configuring the image name to the remoteproc module.
+| ``echo zephyr.elf > /sys/class/remoteproc/remoteproc18/firmware``
+|
+| Once the image name is configured, send the start command.
+| ``echo start > /sys/class/remoteproc/remoteproc18/state``
```

Comment:
At one point, we tried to provide /dev/bone aliases for the various processors to make them consistent. `remoteproc18` isn't guaranteed to stay consistent.